sys/posix/pthread/include/pthread_once.h

```diff
@@ -23,15 +23,21 @@ extern "C" {
 
 /**
  * @brief           Datatype to supply to pthread_once().
+ * @details         This data type must be compatible with the one defined
+ *                  in newlib's `include/sys/_pthreadtypes.h`.
  */
-typedef volatile int pthread_once_t;
+typedef struct {
+  int   is_initialized;  /**< initialized */
```

Comment:
A variable of type `pthread_once_t` can be used by both user code and the gcc/g++ libraries to ensure that the specified function (the init_routine) is executed only once. However, the value of such a `pthread_once_t` variable must only be touched by the `pthread_once()` function. The state of a `pthread_once_t` variable should not be of interest for code usage. 

Before a `pthread_once_t` variable is used, it must be initialized with `PTHREAD_ONCE_INIT`. Since the variable was previously a simple `int` variable, it would be theoretically possible for the user code to initialize it with `= 0`.But that should now trigger a compilation error.